src/generate/templates/header.rs

```diff
@@ -0,0 +1,177 @@
+#[derive(Debug)]
+pub enum ParseError<T> {
+    TooShort(T),
+    NoParse,
+}
+
+pub type ParseResult<'a, 'i, I, T> =
+    Result<Handle<'a, 'i, I, T>, ParseError<Handle<'a, 'i, I, T>>>;
+
+pub type Any = dyn any::Any;
+
+#[derive(Debug)]
+pub struct Ambiguity<T>(T);
+
+pub struct Handle<'a, 'i: 'a, I: 'a + ::gll::runtime::Input, T: ?Sized> {
```

Comment:
I kind of wish all of this stuff could be in the `gll` crate (or a `gll-runtime` crate) and parametrized by `_P` and `_C`, with local aliases.

I wonder if coherence would let us do it the tedious way.

But also, there's no reason we need both `_P` and `_C`, one of them (or a third type) could implement a trait to get the other.
Maybe `I` could be combined into this somehow? As `_P, _C, I` are the 3 type parameters of the `Parser`.